client/src/components/SongCard.jsx

```diff
@@ -89,8 +85,7 @@ export default function SongCard({ title, artist, path_to_song, artistImageUrl,
           variant="subtitle1"
           onClick={(event) => {
             event.stopPropagation();
-            handleSetArtistSongData({ artist }, null);
-            changeView('profile');
+            handleSetArtistSongData(artist, null);
```

Comment:
had to move changeView into the top level function since the view was changing before asynchronous data was returned and state was set